packages/swing-store/test/test-exportImport.js

```diff
@@ -183,6 +191,12 @@ async function testExportImport(
   });
   const { kernelStorage, debug } = ssOut;
 
+  const [bundleID1, bundle1] = await embundle('./faux-module.js');
```

Comment:
One surprising property of the EndoZipBas64 flavor of `bundleSource` is that it bundles packages, not just source files. `./faux-module.js` is a module of the `swing-store` package, and the bundle's `compartment-map.json` will include a compartment record for everything in the transitive dependency graph.

When I went to build support for `b0-` bundles (#7190) on top of this, I happened to add a dependency on Nat (`@endo/nat`). That changed the dependency set, changing the compartment map, which changed the generated bundle IDs.

This test doesn't hard-code any actual bundle IDs, so that wasn't an immediate problem. *But* the tests at line 240-242:
```js
  t.deepEqual(exportData, feedData);
  t.deepEqual(exportData, [
    [`bundle.${bundleID1}`, `${bundleID1}`],
    [`bundle.${bundleID2}`, `${bundleID2}`],
```

embed some assumptions about the sort order of the generated bundle IDs, which breaks 50% of the time.

The original `exportData` is sorted according to whatever the DB returns: the missing `ORDER BY` in `bundleStore.js > sqlGetBundleIDs` means the order will vary depending on what SQLite feels like doing (so we should still add that ORDER BY). Then the test sorts `exportData` lexicographically.

The original `feedData` is ordered chronologically, more or less, since it's built by pushing entries each time the `exportCallback` is run. Then it gets sorted lexicographically too.

The `t.deepEqual(exportData, feedData)` is comparing two equally-sorted lists, no problem.

Then the `t.deepEqual(exportData, [ EXPLICIT ENTRIES ...])` is comparting a lexicographically sorted list against a list of entries that was manually sorted into the same order, *except* that the `bundle.${bundleID1}` and `bundle.${bundleID2}` entries are not sorted. The test assumes that `bundleID1` is always earlier than `bundleID2`, but for any given pair of bundles, that only happens 50% of the time. And the bundles change every time the dependency set changes (or simply their version numbers). So e.g. when we go to release agoric-sdk, the versions will change, and this test has a 50/50 chance of failing.

The most stable fix is to move both `faux-module.js` and `bohr-module.js` into their own package directories, with their own tiny `package.json`. Endo does this in the `compartment-mapper` tests, e.g. https://github.com/endojs/endo/tree/master/packages/compartment-mapper/test/fixtures-cthuloops is a self-contained package (empty `dependencies:`) whose generated bundle will not depend upon any versions of the dependencies of the package doing the tests.

Another fix would be to compare `bundleID1 > bundleID2` after generating both bundles, then assign them to `bundleIDA` and `bundleIDB` in sorted order. Then, in the comparison against `exportData`, use `bundle.${bundleIDA}` instead of ID1, same for IDB/ID2. You'd need to do the same in the comparison of `artifactNames` later: we can't assume that ID1 comes before ID2, but we *can* rely upon IDA coming before IDB.

Reply:
Here's what the second approach would look like:
```diff
diff --git a/packages/swing-store/test/test-exportImport.js b/packages/swing-store/test/test-exportImport.js
index 1b624de25..b2bd2c0fd 100644
--- a/packages/swing-store/test/test-exportImport.js
+++ b/packages/swing-store/test/test-exportImport.js
@@ -197,6 +197,8 @@ async function testExportImport(
   kernelStorage.bundleStore.addBundle(bundleID1, bundle1);
   kernelStorage.bundleStore.addBundle(bundleID2, bundle2);
 
+  let [bundleIDA, bundleIDB] = [bundleID1, bundleID2].sort();
+
   const vats = [
     { vatID: 'vatA', endPos: 0 },
     { vatID: 'vatB', endPos: 0 },
@@ -239,8 +241,8 @@ async function testExportImport(
 
   t.deepEqual(exportData, feedData);
   t.deepEqual(exportData, [
-    [`bundle.${bundleID1}`, `${bundleID1}`],
-    [`bundle.${bundleID2}`, `${bundleID2}`],
+    [`bundle.${bundleIDA}`, `${bundleIDA}`],
+    [`bundle.${bundleIDB}`, `${bundleIDB}`],
     ['kv.brigadoon', 'here during 16'],
     ['kv.kval', 'set in 16'],
     ['kv.vatA.monotonic.10', 'more and more'],
@@ -298,8 +300,8 @@ async function testExportImport(
   ]);
 
   expectedArtifactNames = Array.from(expectedArtifactNames);
-  expectedArtifactNames.push(`bundle.${bundleID1}`);
-  expectedArtifactNames.push(`bundle.${bundleID2}`);
+  expectedArtifactNames.push(`bundle.${bundleIDA}`);
+  expectedArtifactNames.push(`bundle.${bundleIDB}`);
 
   const artifactNames = [];
   for await (const name of exporter.getArtifactNames()) {
```

Reply:
we agreed to rely on the ID sorting, instead of moving the modules into a isolated local package